Sources/Jetworking/Client/Interceptor/Protocol/Interceptor.swift

```diff
@@ -1,4 +1,44 @@
 import Foundation
 
-/// Base Protocol for `RequestInterceptor` and `ResponseInterceptor`.
-public protocol Interceptor {}
+/// Base Protocol for `Interceptor`.
+/// `Interceptors` are able to intercept a request and a response..
+public protocol Interceptor {
+    /**
+     * # Summary
+     * Intercepting the request to create the ability to modify the `URLRequest` before executing it. Example usages are adding request headers, authentication
+     * and request logging.
+     *
+     * - Parameter request:
+     *  The request to be intercepted.
+     *
+     * - Returns:
+     * The intercepted and modified request.
+     */
+    func intercept(_ request: URLRequest) -> URLRequest
+
+    /**
+     * # Summary
+     * Intercepting the response to add the ability to modify a response of a request. Additionally it is possible handle data and errors.
+     *
+     * - Parameter data:
+     *  The data returned by the data task.
+     * - Parameter response:
+     *  The response returned by the data task
+     * - Parameter error:
+     *  The error returned by the data task.
+     *
+     * - Returns:
+     * The intercepted and modified response.
+     */
+    func intercept(data: Data?, response: URLResponse?, error: Error?) -> URLResponse?
+}
+
+extension Interceptor {
+    public func intercept(_ request: URLRequest) -> URLRequest {
+        return request
+    }
+
+    public func intercept(data: Data?, response: URLResponse?, error: Error?) -> URLResponse? {
```

Comment:
Can we swap the argument position of `data` and `response`? If so maybe also remove the parameter name requirement to match the request api.

Reply:
Good point!